src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,115 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const filter = useAppSelector(state => state.filter);
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const filteredTodos: Todo[] | undefined = useMemo(() => {
```

Comment:
```suggestion
  const filteredTodos: Todo[] = useMemo(() => {
```
let's always return array of Todos or empty array